default/python/AI/FleetUtilsAI.py

```diff
@@ -68,107 +68,97 @@ def get_targeted_planet_ids(planet_ids, mission_type):
     return targeted_planets
 
 
-def get_fleets_for_mission(nships, target_stats, min_stats, cur_stats, species, systems_to_check, systems_checked,
-                           fleet_pool_set, fleet_list, take_any=False, extend_search=True,
-                           verbose=False, depth=0):
-    """
-    Implements breadth-first search through systems
-    mutates cur_stats with running status, systems_to_check and systems_checked as systems are checked,
-    fleet_pool_set as fleets are checked also mutates fleet_list as running list of selected fleets;
-    this list will be returned as the function return value if the target statsare met or if upon exhausting
-    systems_to_check both take_any is true and the min stats are met.
+# TODO: as used, cur_stats is  just another return value
+def get_fleets_for_mission(target_stats, min_stats, cur_stats, starting_system,
+                           fleet_pool_set, fleet_list, species=""):
+    """Get fleets for a mission.
+
+    Implements breadth-first search through systems starting at the **starting_sytem**.
```

Comment:
Seems like a good and reasonable change for the future but beyond the scope of this PR.